stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/RetryStreamOperator.java

```diff
@@ -0,0 +1,85 @@
+import uk.gov.justice.services.eventsourcing.repository.jdbc.exception.OptimisticLockingRetryException;
+import uk.gov.justice.services.eventsourcing.source.core.EventSourceTransformation;
+import uk.gov.justice.services.eventsourcing.source.core.EventStream;
+import uk.gov.justice.services.eventsourcing.source.core.exception.EventStreamException;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+
+@ApplicationScoped
+public class RetryStreamOperator {
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    @Value(key = "maxRetryStreamOperation", defaultValue = "10")
+    private long maxRetry;
+
+    @Inject
+    private EventSourceTransformation eventSourceTransformation;
+
+    public void appendWithRetry(final UUID streamId, final EventStream eventStream, final Stream<JsonEnvelope> events) throws EventStreamException {
+        boolean operationCompletedSuccesfully = false;
+        long retryCount = 0L;
+        while (!operationCompletedSuccesfully) {
+            try {
+                eventStream.append(events);
+                logger.info("Appended events to stream with ID - '{}'", streamId);
+                operationCompletedSuccesfully = true;
+            } catch (OptimisticLockingRetryException e) {
+                retryCount++;
+                if (retryCount >= maxRetry) {
+                    logger.error("Failed to append to stream {} due to concurrency issues. Exhausted all retries.", streamId);
+                    throw e;
+                }
+                sleep();
+                logger.warn(format("Encountered exception whilst appending to stream during attempt %s for stream with ID: %s", retryCount, streamId), e);
+            }
+        }
+
+    }
+
+    public void cloneWithRetry(final UUID streamId) throws EventStreamException {
+        boolean operationCompletedSuccesfully = false;
+        long retryCount = 0L;
+        while (!operationCompletedSuccesfully) {
+            try {
+                final UUID clonedStreamId = eventSourceTransformation.cloneStream(streamId);
+                logger.info("Created backup stream '{}' from stream '{}'", clonedStreamId, streamId);
+                operationCompletedSuccesfully = true;
+            } catch (OptimisticLockingRetryException e) {
+                retryCount++;
+                if (retryCount >= maxRetry) {
+                    logger.error("Failed to create backup of stream {} due to concurrency issues. Exhausted all retries.", streamId);
+                    throw e;
+                }
+                sleep();
+                logger.warn(format("Encountered exception whilst cloning stream during attempt %s for stream with ID: %s", retryCount, streamId), e);
+            }
+        }
+
+    }
+
+    private void sleep() {
```

Comment:
We have a class called 'Sleeper' somewhere (framework or event-store) that would allow this to be mocked (and so verified)

Reply:
done